### PR TITLE
[14.0][IMP] fcd_purchase_order: new report product/lot

### PR DESCRIPTION
**Pull request by ChristianSantamaria**
New report product/lot barcodes

### CODE REVIEW

File: fcd_purchase_order/reports/fcd_purchase_order_tag_report.xml [hunk 1]
```diff
@@ -16,6 +16,20 @@
         <field name="dpi">90</field>
     </record>
 
+    <record id="paperformat_cigurria_purchasae_order_product_lot_tag" model="report.paperformat">
+        <field name="name">Purchase Line Cigurria Product/Lot</field>
```
Comment (dalonsod) [inline fcd_purchase_order/reports/fcd_purchase_order_tag_report.xml]: Please remove such reference to customer
```suggestion
        <field name="name">Purchase Line Product/Lot</field>
```

File: fcd_purchase_order/reports/fcd_purchase_order_tag_report.xml [hunk 2]
```diff
@@ -26,4 +40,16 @@
         report_type="qweb-pdf"
         attachment_use="False"
     />
+
+    <report
+        id="action_report_fcd_purchasae_order_product_lot_pdf"
+        model="purchase.order"
+        string="Product/Lot"
```
Comment (dalonsod) [inline fcd_purchase_order/reports/fcd_purchase_order_tag_report.xml]: Better this:
```suggestion
        string="Product/Lot Codes"
```

File: fcd_purchase_order/reports/fcd_purchase_order_tag_template.xml [hunk 3]
```diff
@@ -59,6 +59,35 @@
     </div>
   </template>
 
+  <template id="report_fcd_purchasae_order_product_lot_document_pdf">
+    <t t-set="doc" t-value="doc.with_context(lang=doc.partner_id.lang)"/>
```
Comment (dalonsod) [inline fcd_purchase_order/reports/fcd_purchase_order_tag_template.xml]: This report is for internal usage, so we shouldn't use vendor language IMO

File: fcd_purchase_order/reports/fcd_purchase_order_tag_template.xml [hunk 4]
```diff
@@ -59,6 +59,35 @@
...
+    <div class="page" style="font-size: 18px;">
+      <div class="d-flex">
+        <h2 class="col-8"><t t-esc="doc.name"/></h2>
+        <h3 class="col-4"><t t-esc="doc.date_approve.strftime('%d/%m/%Y')"/></h3>
+      </div>
+      <div class="d-flex">
+        <h3 class="col-12"><t t-esc="doc.partner_id.name"/></h3>
+      </div>
+
+      <t t-foreach="doc.order_line.filtered(lambda x: x.fcd_lot_id.id != False)" t-as="line">
```
Comment (dalonsod) [inline fcd_purchase_order/reports/fcd_purchase_order_tag_template.xml]: Simplify code good practice (not tested):
```suggestion
      <t t-foreach="doc.order_line.filtered(lambda x: x.fcd_lot_id)" t-as="line">
```

**Review by dalonsod — CHANGES_REQUESTED** [review]
See comments. And also, please confirm with LR: should we avoid page breaks like this?

imagen

**Comment by ChristianSantamaria** [comment]
All done and updated test enviroment 👍

**Review by dalonsod — APPROVED** [review]

**Pull request merged**